Clarify getting a receipt is async registration (#198)
ietf-wg-scitt/draft-ietf-scitt-architecture: draft-ietf-scitt-architecture.md

  Custom Signed Statements are evaluated given the current Transparency Service state and the entire Envelope, and may use information contained in the attributes of named policies.
1. **Register the Signed Statement** to the append-only log.
1. **Return the Receipt**, which MAY be asynchronous from registration.

> When asynchronous, the Transparency Service MUST return an identifier used to retrieve the Receipt.

Reviewer: Where is this identifier defined? What does it identify? There is a potential ambiguity between identifying a registration event (later authenticated by the receipt) or more loosely just a signed statement, which may then be registered later, or refer to an earlier registration, possibly subsumed by others. Unless/until we agree on registration-event identifiers, I suggest we cut this sentence.

Reviewer 2: in an asynchronous flow like when batch writing, you need an identifier to use to retrieve the receipt when it eventually becomes available,.

Reviewer: The simpler way to prevent confusion is to remove this sentence. Why is it required in the arch?

Reviewer: Could we write "correlator" instead of identifier to side-step this issue?

Reviewer 3: Could we fiip the sense and instead say

```suggestion
The Transparency Service MUST be able to provide a receipt for all registered statements. A receipt for a signed statement MAY be provided asyncrhously```

I think this just brings what has been said previously closer together but the adjacency makes it easire to digest.

This could be tightened by adding

"Asynchronously obtained receipts MUST be demonstrably equivelent to a receipt obtained at the point of registration"

Reviewer 3: I think this just brings what has been said previously closer together but the adjacency makes it easire to digest.

This could be tightened by adding

"Asynchronously obtained receipts MUST be demonstrably equivelent to a receipt obtained at the point of registration"

> When asynchronous, the Transparency Service MUST returns an identifier used to retrieve the Receipt.

Reviewer 4: Minor grammar issue, but worth fixing:

```suggestion
When asynchronous, the Transparency Service MUST return an identifier used to retrieve the Receipt.
```

> The Transparency Service MUST be able to provide a receipt for all registered statements. A receipt for a signed statement MAY be provided asyncrhously```

Reviewer 4: Minor capitalization and spelling issues, but not blocking

```suggestion
The Transparency Service MUST be able to provide a receipt for all registered Statements. A receipt for a Signed Statement MAY be provided asynchronously```
```

> The Transparency Service MUST be able to provide a receipt for all registered Statements. A receipt for a Signed Statement MAY be provided asynchronously.

Author: ```suggestion
The Transparency Service MUST be able to provide a receipt for all registered Statements.
A receipt for a Signed Statement MAY be provided asynchronously.
```

Details about generating Receipts are described in {{Receipt}}.

The last two steps may be shared between a batch of Signed Statements recorded in the Append-only Log.

> A Transparency Service MUST ensure that a Signed Statement is registered before releasing its Receipt.

Reviewer 2: this seems good

The same Signed Statement may be independently registered in multiple Transparency Services, producing multiple, independent Receipts.

> The multiple receipts may be attached to the unprotected header of the Signed Statement, creating a Transparent Statement.

Reviewer 3: I think this cements the relationship between signed statements and transparent statements and makes the 'recursive' case more natural to think about
